lib/editors/common/shell.js

```diff
@@ -0,0 +1,90 @@
+var shellQuote = require('shell-quote');
+var shellEscape = require('command-join');
+var assign = require('../../utils').assign;
+
+/**
+ * Build a shell command from arguments.
+ *
+ * It correctly escapes the arguments when it's converted to a string.
+ *
+ * Example:
+ *
+ *    shell('vim', 'the code.txt').toString() -> "vim 'the code.txt'"
+ *
+ * API:
+ *
+ *    shell('ls -l')
+ *    shell('ls', '-l')
+ *    shell('sh -c', shell('ls -l'))
+ *
+ * It also supports an argument parameters. It could be useful if it's needed to use some
+ * variables in the command and be ensured what the escaping is correctly applied to them values.
+ *
+ * Example:
+ *    shell('ls -l {dir}').withParams({dir: '/root'})
+ *    shell('sh -c', shell('ls -l {dir}')).withParams({dir: '/root'})
+ *
+ * Arguments are variable number of strings or shell objects
+ */
+function shell(...args) {
+  return new ShellCommand(args);
+}
+
+function ShellCommand(args) {
+  this._args = ShellCommand.parse(args);
```

Comment:
Does we really need make `args` and `params` fields private? Looks like no reasons for this.